### ISSUE TITLE
`zef test Foo` should fetch distribution and dependencies if they are needed

### ISSUE THREAD

**Issue by azawawi** [ugexe/zef#81]
It would be interesting to `zef test Some::Module` without installing it. Currently it only works on paths.

**Comment by ugexe** [comment]
This functionality was both added and removed in the last month.

At first thought it seemed ideal. However it ended up having some less than ideal traits:
* Since `zef test Some::Module` "just worked", then `zef build Some::Module` should also work
* The above `zef build Some::Module` actually did work. But now there was an abundance of redundancy where `zef build Some::Module` `zef test Some::Module` and `zef install Some::Module` all ran the same stuff *except* for the processes after it (`build` stopped right before `test`, `test` stopped right before `install`)
* Since both `test` and `build` were able to accept module names this gave the impression that they should optionally fetch any required dependencies. This also means that both `test` and `build` commands need all the additional options/flags in `multi MAIN` to support search/depends/etc, which in turn makes the automatic `USAGE` that shows commands + options (not the --help `USAGE`) even larger (mostly with redundant flags)
* `test` should not imply running `build`. So the `install` workflow cannot be used as is, adding only a way to stop after testing (it would have to use a different workflow to skip `build`, unlike install)

I'm open to being convinced otherwise, but for now my evidence suggests the pros do not outweight the cons of added redundancy/complexity. The desired behavior can, however, be accomplished a few different ways:

    # Fetch dependencies
    zef --dry install Some::Module # Additionally runs `build` as needed
    zef look Some::Module && zef --depsonly install . && zef test .

    # Don't fetch dependencies
    zef look Some::Module && zef build . && zef test . # Additionally runs `build`
    zef look Some::Module && zef test .

tl;dr: If rakudo's `&MAIN_HELPER` ever gets updated to gracefully handle universal options (options that all `multi MAIN` share) the complexity will no longer be an issue, because it won't have to duplicate the same descriptive (so sometimes longer) variable names in every signature to get the proper `USAGE`

**Comment by azawawi** [comment]
In my mind, `zef test .` depends on a on `zef build .`. Ofcourse one can add a `--no-build` for people who may want that behavior.

**Comment by azawawi** [comment]
Any update on this one?

**Comment by ugexe** [comment]
I will probably bring it back when I have some free time. I see there is a way to hide stuff from `USAGE` that could alleviate my previously mentioned concerns, but because `zef --dry install .` at least provides the functionality already, it's not yet at the top of my todo list

**Comment by azawawi** [comment]
So shall i close it or not?

**Comment by ugexe** [comment]
You can leave it open. Maybe someone else will have an opinion on it before I get around to redoing USAGE
